src/accelerate/accelerator.py

```diff
@@ -499,81 +527,116 @@ def _prepare_deepspeed(self, *args):
+            raise ValueError(
+                "You can only specify `accelerate.utils.DummyScheduler` in the code when using "
+                "`accelerate.utils.DummyOptim`."
             )
+
+        if model is not None:
+            model_parameters = None
+            if isinstance(optimizer, (DummyOptim)):
+                model_parameters = optimizer.param_groups
+                kwargs = dict(model=model, model_parameters=model_parameters, config_params=self.deepspeed_config)
+            else:
+                kwargs = dict(model=model, optimizer=optimizer, config_params=self.deepspeed_config)
+                if type(scheduler).__name__ in deepspeed.runtime.lr_schedules.VALID_LR_SCHEDULES:
+                    kwargs["lr_scheduler"] = scheduler
+
+            engine, optimizer, _, lr_scheduler = deepspeed.initialize(**kwargs)
+            optimizer = DeepSpeedOptimizerWrapper(optimizer)
+            if lr_scheduler is None:
+                scheduler = AcceleratedScheduler(
+                    scheduler,
+                    optimizer,
+                    step_with_optimizer=self.step_scheduler_with_optimizer,
+                    split_batches=self.split_batches,
+                )
+            else:
+                scheduler = DeepSpeedSchedulerWrapper(lr_scheduler, optimizer)
+
             for i in range(len(result)):
                 if isinstance(result[i], torch.nn.Module):
                     result[i] = engine
-                elif isinstance(result[i], torch.optim.Optimizer):
-                    result[i] = DeepSpeedOptimizerWrapper(engine.optimizer, engine)
-            self.deepspeed_engine = engine  # pointing for deepspeed_engine.backward()
+                elif isinstance(result[i], (torch.optim.Optimizer, DummyOptim)):
+                    result[i] = optimizer
+                elif (isinstance(result[i], (torch.optim.lr_scheduler._LRScheduler, DummyScheduler))) or (
+                    type(result[i]).__name__ in deepspeed.runtime.lr_schedules.VALID_LR_SCHEDULES
+                ):
+                    result[i] = scheduler
+            self.deepspeed_engine_wrapped = DeepSpeedEngineWrapper(
+                engine
+            )  # pointing for deepspeed_engine_wrapped.backward()
             self._models.append(engine)
-            self._optimizers.append(engine.optimizer)
+            self._optimizers.append(optimizer)
+            self._schedulers.append(scheduler)
             assert (
                 len(self._models) == 1
             ), "You can't use same `Accelerator()` instance with 2 models when using DeepSpeed"
 
         if self.distributed_type == DistributedType.DEEPSPEED:
             assert hasattr(
-                self, "deepspeed_engine"
-            ), "You need to pass the model along the optimizer when using Deepspeed."
+                self, "deepspeed_engine_wrapped"
+            ), "You need to pass the model along with the optimizer when using Deepspeed."
```

Comment:
Let's take this PR as an opportunity to remove those assers and use proper exceptions :-)